numpy/lib/tests/test_arraysetops.py

```diff
@@ -125,6 +128,60 @@ def test_ediff1d(self):
         assert_array_equal([7,1], ediff1d(two_elem, to_begin=7))
         assert_array_equal([5,6,1], ediff1d(two_elem, to_begin=[5,6]))
 
+    @pytest.mark.parametrize("ary, prepend, append", [
+        (np.array([1, 2, 3], dtype=np.int64),
+         None,
+         np.nan),
+        (np.array([1, 2, 3], dtype=np.int32),
+         np.array([5, 7, 2], dtype=np.int64),
+         None),
+        (np.array([1., 3., 9.], dtype=np.int8),
+         np.nan,
+         np.nan),
+         ])
+    def test_ediff1d_type_cast(self, ary, prepend, append):
```

Comment:
Nit: The test name should really indicate that these are invalid /unsafe / forbidden type casts